coreneuron/utils/reports/report_event.cpp

```diff
@@ -0,0 +1,43 @@
+#include "report_event.h"
+#include "coreneuron/nrnoc/multicore.h"
+#include "coreneuron/nrniv/nrn_assert.h"
+#ifdef ENABLE_REPORTING
+#include "bbp/sonata/reports.h"
+#include "reportinglib/Records.h"
+#endif  // ENABLE_REPORTING
+
+namespace coreneuron {
+
+#ifdef ENABLE_REPORTING
+ReportEvent::ReportEvent(double dt, double tstart, VarsToReport &filtered_gids, const char *name)
+        : dt(dt), tstart(tstart) {
+    strcpy(report_path, name);
+    VarsToReport::iterator it;
+    nrn_assert(filtered_gids.size());
+    step = tstart / dt;
+    gids_to_report.reserve(filtered_gids.size());
+    for (it = filtered_gids.begin(); it != filtered_gids.end(); ++it) {
+        gids_to_report.push_back(it->first);
+    }
+    std::sort(gids_to_report.begin(), gids_to_report.end());
+}
+
+/** on deliver, call ReportingLib and setup next event */
+void ReportEvent::deliver(double t, NetCvode *nc, NrnThread *nt) {
```

Comment:
If it doesn't break the API, `nc` and `nt` should be references instead of pointers if there are never `nullptr`.